src/components/Popups/ReplyField/ReplyFormContainer.tsx

```diff
@@ -0,0 +1,80 @@
+import { connect } from 'react-redux';
+import { ContentState, EditorState, SelectionState } from 'draft-js';
+import { FormikBag, withFormik } from 'formik';
+import ReplyForm from './ReplyForm';
+import withMentionDecorator from './withMentionDecorator';
+import { AppState, getCreatorCursor } from '../../../store';
+
+const MAX_LENGTH = 10000;
+
+export type ReduxProps = {
+    cursorPosition: number;
+};
+
+export type Props = {
+    isPending: boolean;
+    onCancel: (text: string) => void;
+    onChange: (text: string) => void;
+    onSubmit: (text: string) => void;
+    value?: string;
+} & ReduxProps;
+
+export type FormErrors = {
+    editorState?: string;
+};
+
+export type FormValues = {
+    editorState: EditorState;
+};
+
+export const mapStateToProps = (state: AppState): ReduxProps => ({
+    cursorPosition: getCreatorCursor(state),
+});
+
+export const mapPropsToValues = ({ cursorPosition: prevCursorPosition, value = '' }: Props): FormValues => {
+    const contentState = ContentState.createFromText(value as string);
+    const cursorPosition = value ? prevCursorPosition : 0;
+    let editorState = withMentionDecorator(EditorState.createWithContent(contentState));
+
+    editorState = EditorState.forceSelection(
```

Comment:
This variable is using itself to mutate itself, which seems a little weird. Should we break it up?

```
const contentState = ContentState.createFromText(value as string);
const mentionState = withMentionDecorator(EditorState.createWithContent(contentState));

return {
    editorState: EditorState.forceSelection(
        mentionState,
        mentionState.getSelection().merge({
            anchorOffset: cursorPosition,
            focusOffset: cursorPosition,
            hasFocus: true,
        }) as SelectionState,
    );